forem/forem
# Save selected tags to the display ad model

## What type of PR is this? (check all applicable)


## Description
In the [previous PR](https://github.com/forem/forem/pull/18560), we integrated the multi select autocomplete component, however we were not saving the selected tags to the database. Hence, in this PR, we save the selected tags to the database. and ensure that they display correctly on edit as well.

I'm building the functionality incrementally to keep the PR as small as possible and incrementally release behind a feature flag. Hence, in the next PR's I will work on:

- [ ]  Ensure tag list is maximally optimized to replicate what we have done on articles. We have really optimized the queries on articles, and we should copy all of that over to DisplayAds. This will involve adding a new cache column and index in addition to the drop-in-place ActsAsTaggableOn functionality.
- [ ]  Ensuring that the Tags field only shows up when we choose "Below the comment section"
- [ ]  Ensure that the Display Ad should appear on Posts with at least one matching Tag and if there are multiple tags we round robin them.
- [ ] Add some cypress tests if necessary

## Related Tickets & Documents
- Related Issue #https://github.com/forem/forem/issues/18519

## QA Instructions, Screenshots, Recordings

See loom video for demo:
https://www.loom.com/share/ca06213eac1144fa9133a53bbea78d53
(dunno why Loom is annotating "tags" as "ticks" - maybe it's my accent 🤷🏽‍♀️)

- Create a Display Ad with some tags and ensure they're getting saved to the database.
- Disable the Feature Flag and ensure that everything still works as expected. 

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [x] Yes on the Rails side, but I'll add some more tests to the JS with Cypress later on.
      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](https://media.giphy.com/media/LVRBKMg4H33bcRZ3IQ/giphy.gif)
